### ISSUE TITLE
Precision-Recall Curve: switch axes in plot

### ISSUE THREAD

**Issue by mykappa** [Lightning-AI/torchmetrics#2161]
## 🚀 Feature

In the precision-recall curve plot, place recall on the x-axis and precision on the y-axis. 

### Motivation

In my experience, precision-recall curves with precision on the y-axis (ordinate) and recall on the x-axis (abscissa) are much more common than the other way around (i.e., how the plot is generated at the moment). A quick Google image search confirms this. And it is also the case in the [sklearn implementation](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.PrecisionRecallDisplay.html).

### Pitch

Place recall on the x-axis and precision on the y-axis in `PrecisionRecallCurve.plot()`

### Alternatives

Add an option to choose between both variants. 

### Additional context

Furthermore, an option for plotting the chance value similar to sklearn's `plot_chance_level` argument would be a welcome addition to the precision-recall curve plot.

**Comment by Yann-CV** [comment]
@SkafteNicki for info this linked to https://github.com/Lightning-AI/torchmetrics/issues/2200
